tensorboard/plugins/interactive_inference/witwidget/notebook/colab/wit.py

```diff
@@ -234,59 +202,10 @@ def _get_element_html(self):
     return """
       <link rel="import" href="/nbextensions/wit-widget/wit_jupyter.html">"""
 
-  def _set_examples(self, examples):
-    self.examples = [json_format.MessageToJson(ex) for ex in examples]
-    self.updated_example_indices = set(range(len(examples)))
-
-  def json_to_proto(self, json):
-    ex = (tf.train.SequenceExample()
-          if self.config.get('are_sequence_examples')
-          else tf.train.Example())
-    json_format.Parse(json, ex)
-    return ex
-
   def infer(self):
-    indices_to_infer = sorted(self.updated_example_indices)
-    examples_to_infer = [
-        self.json_to_proto(self.examples[index]) for index in indices_to_infer]
-    infer_objs = []
-    serving_bundle = inference_utils.ServingBundle(
-      self.config.get('inference_address'),
-      self.config.get('model_name'),
-      self.config.get('model_type'),
-      self.config.get('model_version'),
-      self.config.get('model_signature'),
-      self.config.get('uses_predict_api'),
-      self.config.get('predict_input_tensor'),
-      self.config.get('predict_output_tensor'),
-      self.estimator_and_spec.get('estimator'),
-      self.estimator_and_spec.get('feature_spec'),
-      self.custom_predict_fn)
-    infer_objs.append(inference_utils.run_inference_for_inference_results(
-        examples_to_infer, serving_bundle))
-    if ('inference_address_2' in self.config or
-        self.compare_estimator_and_spec.get('estimator') or
-        self.compare_custom_predict_fn):
-      serving_bundle = inference_utils.ServingBundle(
-        self.config.get('inference_address_2'),
-        self.config.get('model_name_2'),
-        self.config.get('model_type'),
-        self.config.get('model_version_2'),
-        self.config.get('model_signature_2'),
-        self.config.get('uses_predict_api'),
-        self.config.get('predict_input_tensor'),
-        self.config.get('predict_output_tensor'),
-        self.compare_estimator_and_spec.get('estimator'),
-        self.compare_estimator_and_spec.get('feature_spec'),
-        self.compare_custom_predict_fn)
-      infer_objs.append(inference_utils.run_inference_for_inference_results(
-          examples_to_infer, serving_bundle))
-    self.updated_example_indices = set()
-    inferences = {
-      'inferences': {'indices': indices_to_infer, 'results': infer_objs},
-      'label_vocab': self.config.get('label_vocab')}
+    base.WitWidgetBase.infer_impl(self)
     output.eval_js("""inferenceCallback('{inferences}')""".format(
-      inferences=json.dumps(inferences)))
+      inferences=json.dumps(self.inferences)))
```

Comment:
It seems that we duplicate the memory usage of the tool for inference values by keeping it in the class. Is there any advantage to this?

Reply:
fixed